pkg/signature/kms/ehsm/signer.go

```diff
@@ -0,0 +1,197 @@
+package ehsm
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/options"
+)
+
+// nolint:revive
+const (
+	AlgorithmEHRSA4096 = "EH_RSA_4096"
+	AlgorithmEHRSA3072 = "EH_RSA_3072"
+	AlgorithmEHRSA2048 = "EH_RSA_2048"
+	AlgorithmEHECP256  = "EH_EC_P256"
+	AlgorithmEHECP224  = "EH_EC_P224"
+	AlgorithmEHECP384  = "EH_EC_P384"
+	AlgorithmEHECP521  = "EH_EC_P521"
+)
+
+var ehsmSupportedAlgorithms = []string{
+	AlgorithmEHRSA4096,
+	AlgorithmEHRSA3072,
+	AlgorithmEHRSA2048,
```

Comment:
why not use the name EH_RSA_4096 directly?

Reply:
fix